autoload/ale/handlers/haskell_stack.vim

```diff
@@ -0,0 +1,7 @@
+function! ale#handlers#haskell_stack#EscapeExecutable(executable, stack_exec) abort
+    let l:exec_args = a:executable =~? '^stack$'
```

Comment:
This checks for an executable which exactly matches `'stack'`, so this is the same as `is? 'stack'`. Use `=~? 'stack$'` instead so any path ending in `'stack'` will be treated like a stack executable.